doc/source/contributing/docstrings.md

```diff
@@ -0,0 +1,68 @@
+(docstrings)=
+# Docstring formatting and type hints
+
+Docstrings should follow the
+[numpy docstring guide](https://numpydoc.readthedocs.io/en/latest/format.html).
+Extra guidance can also be found in
+[pandas docstring guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing_docstring.html).
+Please reasonably document any additions or changes to the codebase,
+when in doubt, add a docstring.
+
+The different formatting and aim between numpydoc style type description and
+[type hints](https://docs.python.org/3/library/typing.html)
+should be noted. numpydoc style targets docstrings and aims to be human
+readable whereas type hints target function definitions and `.pyi` files and
+aim to help third party tools such as type checkers or IDEs. ArviZ does not
+require functions to include type hints
+however contributions including them are welcome.
+
+## References
+Docstrings follow the same guide for adding references as the other docs.
+For adding refernces to external libraries functions and objects, see {ref}`reference_external_libs`. For referencing ArviZ objects, follow {ref}`reference_arviz_objects`.
+
+## See Also
```

Comment:
The see also section, like all other sections of the docstring is already covered in the numpydoc guide. I would mention the see also guide because for us it is a key section that _must_ be present because we have a lot of interconnected functions both within the library and with external libraries like xarray for example and we have seen that users have a hard time finding the functions they are searching for. But I would not focus too much (or at all) on how to format it as numpydoc already does that, we could for example explain things that are not covered in the numpydoc style guide like how both `arviz.plot_hdi` and `plot_hdi` work and link to the same place but show different text for the hyperlink